diff --git a/mod.d.ts b/mod.d.ts
index b1b5ce7f7de951186c28949ea2980bdf6d4bfed4..035140a2699bee48a12ee5e6661023d48ee68f54 100755
--- a/mod.d.ts
+++ b/mod.d.ts
@@ -1,3 +1,3 @@
 declare const distance: (a: string, b: string) => number;
-declare const closest: (str: string, arr: readonly string[]) => string;
+declare const closest: (str: string, arr: string[] | Set<string>) => string;
 export { closest, distance };
diff --git a/mod.js b/mod.js
index 6bc27459399e6d5a7ef1d991aa8b3764176e36c2..3c38721d4a0f2060db5998ff6c09a18125192c0a 100755
--- a/mod.js
+++ b/mod.js
@@ -130,7 +130,7 @@ exports.distance = distance;
 var closest = function (str, arr) {
     var min_distance = Infinity;
     var min_index = 0;
-    for (var i = 0; i < arr.length; i++) {
+    for (var i = 0; i < arr.length ?? arr.size; i++) {
         var dist = distance(str, arr[i]);
         if (dist < min_distance) {
             min_distance = dist;
